sacred/observers/file_storage.py

```diff
@@ -180,14 +197,23 @@ def started_event(
         return os.path.relpath(self.dir, self.basedir) if _id is None else _id
 
     def find_or_save(self, filename, store_dir: Path):
-        os.makedirs(str(store_dir), exist_ok=True)
-        source_name, ext = os.path.splitext(os.path.basename(filename))
-        md5sum = get_digest(filename)
-        store_name = source_name + "_" + md5sum + ext
-        store_path = store_dir / store_name
-        if not store_path.exists():
-            copyfile(filename, str(store_path))
-        return store_path, md5sum
+        try:
+            Path(filename).resolve().relative_to(Path(self.basedir).resolve())
+            is_relative_to = True
+        except ValueError:
+            is_relative_to = False
+
+        if self.no_duplicate and is_relative_to:
+            return filename
+        else:
+            os.makedirs(str(store_dir), exist_ok=True)
```

Comment:
You already used `pathlib` in the code block above this. Would be nice to use `pathlib` here as well

Reply:
Done.